### ISSUE TITLE
drenv rook-cluster addon: repository 'https://raw.githubusercontent.com/rook/rook/' not found

### ISSUE THREAD

**Issue by nirs** [RamenDR/ramen#1310]
Seen multiple times when deploying rook cluster:

```
drenv.commands.Error: Command failed:
   command: ('addons/rook-cluster/start', 'dr1')
   exitcode: 1
   error:
      Traceback (most recent call last):
        File "/home/nsoffer/ramen/test/addons/rook-cluster/start", line 54, in <module>
          deploy(cluster)
        File "/home/nsoffer/ramen/test/addons/rook-cluster/start", line 27, in deploy
          kubectl.apply("--kustomize", kustomization, context=cluster)
        File "/home/nsoffer/ramen/test/drenv/kubectl.py", line 59, in apply
          _watch("apply", *args, input=input, context=context, log=log)
        File "/home/nsoffer/ramen/test/drenv/kubectl.py", line 157, in _watch
          for line in commands.watch(*cmd, input=input):
        File "/home/nsoffer/ramen/test/drenv/commands.py", line 155, in watch
          raise Error(args, error, exitcode=p.returncode)
      drenv.commands.Error: Command failed:
         command: ('kubectl', 'apply', '--context', 'dr1', '--kustomize', '/tmp/drenvwr8jigk8')
         exitcode: 1
         error:
            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/cluster-test.yaml': Get "https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/cluster-test.yaml": dial tcp: lookup raw.githubusercontent.com on 127.0.0.53:53: read udp 127.0.0.1:39304->127.0.0.53:53: i/o timeout': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 404: Not Found
            fatal: repository 'https://raw.githubusercontent.com/rook/rook/' not found
            : exit status 128
```

The root cause is using kustomization for relatively small resource. kustomize need to fetch the entire rook repository for getting single file. We can avoid this by keeping the cluster resource in ramen source.

The reason we kustomization is to make ramen modification clear (as patches). We can replace the the patches with comments.

## Tasks

- [x] Should be fixed by #1290 - the error will move to cache refresh time instead of build time

**Comment by nirs** [comment]
Another version of this error - this time it is not related to the flaky network, seems like random error on the server side. 

```
drenv.commands.Error: Command failed:
   command: ('addons/rook-cluster/start', 'dr2')
   exitcode: 1
   error:
      Traceback (most recent call last):
        ...
      drenv.commands.Error: Command failed:
         command: ('kubectl', 'apply', '--context', 'dr2', '--kustomize', '/tmp/drenvgnvkwkgc')
         exitcode: 1
         error:
            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/cluster-test.yaml': URL is a git repository': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 500: Internal Server Error
            fatal: unable to access 'https://raw.githubusercontent.com/rook/rook/': The requested URL returned error: 500
            : exit status 128
```

**Comment by nirs** [comment]
This error failed 4 builds of 100 in the last run:

```
$ grep -A8 ' drenv.commands.Error' out/*.log
out/045.log:      drenv.commands.Error: Command failed:
out/045.log-         command: ('kubectl', 'apply', '--context', 'dr1', '--kustomize', '/tmp/drenvdjy0p17o')
out/045.log-         exitcode: 1
out/045.log-         error:
out/045.log-            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
out/045.log-            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/cluster-test.yaml': URL is a git repository': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 404: Not Found
out/045.log-            fatal: repository 'https://raw.githubusercontent.com/rook/rook/' not found
out/045.log-            : exit status 128
out/045.log-
--
out/046.log:      drenv.commands.Error: Command failed:
out/046.log-         command: ('kubectl', 'apply', '--context', 'dr2', '--kustomize', '/tmp/drenvn1_1dfyo')
out/046.log-         exitcode: 1
out/046.log-         error:
out/046.log-            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
out/046.log-            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/crds.yaml': URL is a git repository': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 404: Not Found
out/046.log-            fatal: repository 'https://raw.githubusercontent.com/rook/rook/' not found
out/046.log-            : exit status 128
out/046.log-
--
out/047.log:      drenv.commands.Error: Command failed:
out/047.log-         command: ('kubectl', 'apply', '--context', 'dr1', '--kustomize', '/tmp/drenvfy6xh20x')
out/047.log-         exitcode: 1
out/047.log-         error:
out/047.log-            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
out/047.log-            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/crds.yaml': URL is a git repository': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 404: Not Found
out/047.log-            fatal: repository 'https://raw.githubusercontent.com/rook/rook/' not found
out/047.log-            : exit status 128
out/047.log-
--
out/048.log:      drenv.commands.Error: Command failed:
out/048.log-         command: ('kubectl', 'apply', '--context', 'dr1', '--kustomize', '/tmp/drenvjp0t7_qu')
out/048.log-         exitcode: 1
out/048.log-         error:
out/048.log-            # Warning: 'patchesJson6902' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
out/048.log-            error: accumulating resources: accumulation err='accumulating resources from 'https://raw.githubusercontent.com/rook/rook/release-1.13/deploy/examples/crds.yaml': URL is a git repository': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/rook/rook HEAD': remote: 404: Not Found
out/048.log-            fatal: repository 'https://raw.githubusercontent.com/rook/rook/' not found
out/048.log-            : exit status 128
out/048.log-
```

**Comment by nirs** [comment]
Fixed by #1353
